tpm2/Makefile

```diff
@@ -254,91 +255,124 @@ INDENT_OPTS += -l80#                     max non-comment line length is 80
 #====================== START: RULES =========================================
 
 .PHONY: all
-all: pre kmyth-lib kmyth-seal kmyth-unseal kmyth-getkey
+all: pre clean-backups logger-lib tpm-util-lib \
+     $(BIN_DIR)/kmyth-seal \
+     $(BIN_DIR)/kmyth-unseal \
+     $(BIN_DIR)/kmyth-getkey
 
-kmyth-lib: $(LOGGER_OBJECTS) \
-           $(UTIL_OBJECTS) \
-           $(CIPHER_OBJECTS) \
-           $(TPM_OBJECTS)
+.PHONY: pre
+pre:
+	indent $(INDENT_OPTS) $(SOURCE_FILES)
+	indent $(INDENT_OPTS) $(HEADER_FILES)
+
+.PHONY: logger-lib
+logger-lib: pre  clean-backups $(LOGGER_OBJECTS) | $(LIB_DIR)
 	$(CC) $(SOFLAGS) \
 	      $(LOGGER_OBJECTS) \
+	      -o $(LOGGER_LIB_LOCAL_DEST) \
+	      $(LDLIBS)
+
+.PHONY: tpm-util-lib
```

Comment:
Instead of making the target phony can we add `logger-lib` as a prerequisite so it gets built anytime either one of it's sources changes or the logger changes?

Reply:
Similarly as above, I made $(LIB_DIR)/libkmyth-tpm.so a non-phony target but changed 'tpm-util-lib' to the more accurate and simpler 'libs' phony target, which has $(LIB_DIR)/libkmyth-logger.so and $(LIB_DIR)/libkmyth-tpm.so as pre-requisites.

Now, for *make all* or *make libs*, the libraries will get built based on changes to any pre-reqs. For *make logger-lib*, only the logger library will get built based on changes to its pre-reqs.

I think we want the 'logger-lib' to ignore the pre-reqs to the TPM utilities, because we don't want someone to have deal with TPM issues if all they want is the logger library.